Document TelemetrySource property of the ExecutionRejectedException (#2355)
App-vNext/Polly: docs/strategies/rate-limiter.md

### Failure handling

> At the first glance it might not be obvious what is the difference between these two techniques:

Reviewer: ```suggestion
At first glance it might not be obvious what the difference between these two techniques is:
```

```
<!-- endSnippet -->

## Defaults

| Property                    | Default Value                                        | Description                                                                                                 |
|-----------------------------|------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
| `RateLimiter`               | `null`                                               | **Dynamically** creates a `RateLimitLease` for executions.                                                  |
| `DefaultRateLimiterOptions` | `PermitLimit` set to 1000 and `QueueLimit` set to 0. | If `RateLimiter` is not provided then this options object will be used for the default concurrency limiter. |
| `OnRejected`                | `null`                                               | If provided then it will be invoked after the limiter rejected an execution.                                |

> ### `OnRejected` versus catching `RateLimiterRejectedException`

Author: I've just moved this subsection under the `Failure handling`.

> The `RateLimiterRejectedException` has a `RetryAfter` and a `TelemetrySource` property. If the `RetryAfter` optional `TimeSpan` is provided then this indicates that your requests are throttled and you should retry them no sooner than the value given.

Reviewer: ```suggestion
The `RateLimiterRejectedException` has a `RetryAfter` and a `TelemetrySource` property. If the optional `RetryAfter` value is provided then this indicates that your requests are being throttled and you should retry them no sooner than the property's value.
```

> [!NOTE]

> > Please note that the `RetryAfter` information is not available inside the `OnRejected` callback.

Reviewer: Could you also remove the "please note" prefix from line 158 too?

```suggestion
> The `RetryAfter` value is not available inside the `OnRejected` callback.
```

>  The `TelemetrySource` property is a [`ResilienceTelemetrySource`](xref:Polly.Telemetry.ResilienceTelemetrySource) which allows you retrieve information like the executed pipeline and the executed strategy. These can be really handy whenever you have multiple limiter strategies in your pipeline (for example a rate and a concurrency limiter) and you want to know which strategy threw the `RateLimiterRejectedException`.

Reviewer: ```suggestion
 The `TelemetrySource` property is a [`ResilienceTelemetrySource`](xref:Polly.Telemetry.ResilienceTelemetrySource) which allows you retrieve information such as the executed pipeline and strategy. These can be useful if you have multiple limiter strategies in your pipeline (for example both a rate and concurrency limiter) and you need to know which strategy caused the `RateLimiterRejectedException` to be thrown.
```
